internal/configs/parsing_helpers_test.go

```diff
@@ -617,6 +617,79 @@ func TestParseProxyBuffersSpec(t *testing.T) {
 	}
 }
 
+func TestParseProxyBuffersSpecWithAutoAdjust(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+		hasError bool
+	}{
+		{
+			name:     "valid proxy buffers with k unit",
+			input:    "8 4k",
+			expected: "8 4k",
+			hasError: false,
+		},
+		{
+			name:     "valid proxy buffers with M unit",
+			input:    "32 2M",
+			expected: "32 2M",
+			hasError: false,
+		},
+		{
+			name:     "auto-adjust random letter to m",
+			input:    "16 8x",
+			expected: "16 8m",
+			hasError: false,
+		},
+		{
+			name:     "empty string",
+			input:    "",
+			expected: "",
+			hasError: true,
+		},
+		{
+			name:     "only buffer count",
+			input:    "8",
+			expected: "",
+			hasError: true,
+		},
+		{
+			name:     "negative buffer count",
+			input:    "-8 4k",
+			expected: "",
+			hasError: true,
+		},
+		{
+			name:     "non-numeric buffer size",
+			input:    "8 abc",
+			expected: "",
+			hasError: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := ParseProxyBuffersSpecWithAutoAdjust(tc.input)
+
+			if tc.hasError {
+				if err == nil {
+					t.Errorf("ParseProxyBuffersSpecWithAutoAdjust(%q) expected error but got none, result: %q", tc.input, result)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("ParseProxyBuffersSpecWithAutoAdjust(%q) unexpected error: %v", tc.input, err)
+				}
+				if result != tc.expected {
+					t.Errorf("ParseProxyBuffersSpecWithAutoAdjust(%q) = %q, expected %q", tc.input, result, tc.expected)
+				}
+			}
```

Comment:
Hopefully this makes understanding the logic a little more readable
```suggestion
			if tc.hasError && err == nil {
			  t.Errorf("ParseProxyBuffersSpecWithAutoAdjust(%q) expected error but got none, result: %q", tc.input, result)
            }
		    if err != nil {
			    t.Errorf("ParseProxyBuffersSpecWithAutoAdjust(%q) unexpected error: %v", tc.input, err)
		    }
		    if result != tc.expected {
			    t.Errorf("ParseProxyBuffersSpecWithAutoAdjust(%q) = %q, expected %q", tc.input, result, tc.expected)
		    }
```